src/transport/SessionManager.cpp

```diff
@@ -773,29 +779,60 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
         return;
     }
 
-    System::PacketBufferHandle msgCopy;
+    // Extract MIC from the end of the message.
+    uint8_t * data     = msg->Start();
+    uint16_t len       = msg->DataLength();
+    uint16_t footerLen = packetHeader.MICTagLength();
+    VerifyOrReturn(footerLen <= len); // err = CHIP_ERROR_INVALID_MESSAGE_LENGTH;
+
+    uint16_t taglen = 0;
+    MessageAuthenticationCode mac;
+    ReturnOnFailure(mac.Decode(packetHeader, &data[len - footerLen], footerLen, &taglen));
+    VerifyOrReturn(taglen == footerLen); // err = CHIP_ERROR_INTERNAL;
+
     bool decrypted = false;
     while (!decrypted && iter->Next(groupContext))
     {
+        CryptoContext context(groupContext.keyContext);
+        msgCopy = msg.CloneData();
+
+        if (packetHeader.HasPrivacyFlag())
```

Comment:
I don't see the changes that implement the outgoing part of message transmission using the privacy flag -if that's not present in the tree right now, any reason why that's not bundled with this PR as well?

Reply:
I'd like to get 1.0 nodes to do the right thing on the receive side ASAP.  Sending with Pflag = 1 has extra considerations given SVE and 1.0 nodes before this patch won't be able to process such messages.

Reply:
sending private messages is out-of-scope of this PR.